utils/fileops.go

```diff
@@ -0,0 +1,79 @@
+package utils
+
+import (
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"github.com/coreos/clair/pkg/commonerr"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+)
+
+func isWritable(path string) bool {
+	return unix.Access(path, unix.W_OK) == nil
+}
+
+func IsPathOk(path string) error {
+	stat, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		return errors.Wrapf(err, "path %q does not exists", path)
+	}
+	if !stat.IsDir() {
+		return errors.Wrapf(err, "path %q is not a directory", path)
+	}
+	if !isWritable(path) {
+		return errors.Wrapf(err, "path %q is not writable", path)
+	}
+	return nil
+}
+
+func StoreHTTPResponseToFile(r *http.Response, fileName string, dataFeedName string) error {
+	gr, err := gzip.NewReader(r.Body)
```

Comment:
Are you sure that this body is *always* gzipped? Check the response headers to be sure.